File: Solution Using C#/Fawry.Ecommerce/Service/ShippingService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Fawry.Ecommerce.Service
{
    public static class ShippingService
    {
        public static void Shipping(List<(IShippable t,int count)> items)
        {
            //var totalWeight = products.Sum(p => p.GetWeight());
            //Console.WriteLine("** Shipping the following products **");
            //foreach (var product in products)
            //{
            //    Console.WriteLine(product);
            //}
            //Console.WriteLine($"Total package weight {totalWeight} kg\n\n================\n");
            double totalWeight = 0;
            Console.WriteLine("** Shipment Notice **");
            foreach (var (item, quantity) in items)
            {
                double itemWeight = item.GetWeight() * quantity;
                Console.WriteLine($"{quantity}x {item.GetName()} {itemWeight}g");
                totalWeight += itemWeight;
            }

            if (totalWeight < 1000)
                Console.WriteLine($"Total Package Weight {totalWeight} g\n");
            else
                Console.WriteLine($"Total Package Weight {totalWeight / 1000} kg\n");

        }
    }
}

File: Solution Using Java/src/Service/ShippingService.java
package Service;

import models.IShippable;
import models.ShippableItemWithItemsCount;

import java.util.List;

public class ShippingService {
    public static void shipping(List<ShippableItemWithItemsCount> items){
        double totalWeight = 0;
        System.out.println("** Shipment Notice **");
        for(ShippableItemWithItemsCount item:items){
            IShippable  shippable = item.getItem();
            int count = item.getCount();
            double itemWeight = shippable.getWeight() * count;
            totalWeight += itemWeight;
            System.out.println(count+"x "+shippable.getName()+ " "+ itemWeight+"g");
        }

        if (totalWeight < 1000) {
            System.out.println("Total package weight "+ totalWeight +" g\n");
        }
        else{
            System.out.println("Total package weight "+ (totalWeight/1000) +" kg\n");
        }
    }
}
